sklearn/linear_model/tests/test_base.py

```diff
@@ -461,36 +463,74 @@ def test_preprocess_data_multioutput():
         assert_array_almost_equal(yt, y - y_mean)
 
 
-def test_preprocess_data_weighted():
+@pytest.mark.parametrize("is_sparse", [False, True])
+def test_preprocess_data_weighted(is_sparse):
     n_samples = 200
     n_features = 2
```

Comment:
Let's add more feature to also test edge cases with constant values (see comment below):

```suggestion
    n_features = 4
```